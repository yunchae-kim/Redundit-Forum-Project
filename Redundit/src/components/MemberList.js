import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import groupIcon from "../test_data/groupIcon.jpg";
import { getAllUsersInGroup } from "../utility/axiosUser";
import "../style/components/MemberList.css";

const MemberList = ({ groupId }) => {
  const [members, setMembers] = useState([]);
  const [fetched, setfetched] = useState(false);

  useEffect(() => {
    setfetched(true);
    getAllUsersInGroup(groupId).then((membersFetched) => {
      setMembers(membersFetched.filter((m) => m.pending === 0));
    });
  }, [fetched]);

  const foldedMembers = members.slice(0, 8).map((m) => (
    <div key={m.iduser}>
      <div className="ML-entry">
        <img className="ML-userIcon" src={groupIcon} alt="" />
        <Link to={{ pathname: `/user/${m.iduser}`, state: { userId: m.iduser } }} className="ML-entry-name" key={m.iduser}><span>{m.username}</span></Link>
        {m.admin === 1 && <span className="ML-entry-admin">Administrator</span>}
      </div>
      <hr className="ML-hr" />
    </div>
  ));

  const expandedMembers = members.map((m) => (
    <div key={m.iduser}>
      <div className="ML-entry">
        <img className="ML-userIcon" src={groupIcon} alt="" />
        <Link to={{ pathname: `/user/${m.iduser}`, state: { userId: m.iduser } }} className="ML-entry-name" key={m.iduser}><span>{m.username}</span></Link>
        {m.admin === 1 && <span className="ML-entry-admin">Administrator</span>}
      </div>
      <hr className="ML-hr" />
    </div>
  ));

  const [isFolded, setIsFolded] = useState(true);
  const unfold = () => {
    setIsFolded(false);
  };

  return (
    <div className={isFolded ? "MemberList" : "MemberList-expand"}>
      <div className="ML-header">
        <div className="ML-title">Group Members</div>
      </div>
      <hr className="ML-hr" />
      {isFolded ? foldedMembers : expandedMembers}
      {isFolded && members.length > 8 && <div className="ML-btn-entry"><button type="submit" onClick={unfold} className="ML-btn">View More </button></div>}
    </div>
  );
};

export default MemberList;
