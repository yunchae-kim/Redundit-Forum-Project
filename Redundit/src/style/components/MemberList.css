.MemberList {
    width: 400px;
    background-color: white;
    border-radius: 20px;
    color: black;
    font-size: 15px;
  }
  
  .MemberList-expand {
    width: 400px;
    background-color: white;
    border-radius: 20px;
    color: black;
    font-size: 15px;
  }
  
  .ML-header {
    padding-top: 24px;
    padding-bottom: 24px;
    margin: 0px;
    font-size: 28px;
    font-weight: 600;
    color: #188bd8;
  }
  
  .ML-title {
    margin-top: 0px;
    margin-bottom: 0px;
  }
  
  .ML-entry {
    display: flex;
    justify-items: flex-start;
    align-items: center;
    font-size: 20px;
    margin-top: 16px;
    margin-bottom: 16px;
  }
  
  .ML-hr {
    border: 1px solid #d1e8f7;
    margin-bottom: 0px;
    margin-top: 0px;
  }
  
  .ML-userIcon {
    border-radius: 50%;
    width: 40px;
    height: 40px;
    align-self: center;
    margin-left: 32px;
  }
  
  .ML-entry-admin {
    color: #465771;
    margin-left: 32px;
  }
  
  .ML-entry-name {
    margin-left: 24px;
    color: #111;
  }
  
  .ML-btn-entry {
    display: block;
    font-size: 20px;
  }
  
  .ML-btn {
    color: #5da271;
    padding: 15px;
    font-size: 20px;
    font-weight: 600;
    background-color: transparent;
    border: none;
  }
  
  .ML-manage {
    display: flex;
  }