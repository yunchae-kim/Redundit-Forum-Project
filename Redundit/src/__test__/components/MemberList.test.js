import {
  waitFor,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import MemberList from "../../components/MemberList";

test("test should NavBar match snapshot", async () => {
  const tree = renderer.create(
    <BrowserRouter>
      <MemberList groupId={1} />
    </BrowserRouter>,
  ).toJSON();
  await waitFor(async () => {
    expect(tree).toMatchSnapshot();
  });
});
