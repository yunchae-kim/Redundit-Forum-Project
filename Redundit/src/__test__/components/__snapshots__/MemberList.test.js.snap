// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should NavBar match snapshot 1`] = `
<div
  className="MemberList"
>
  <div
    className="ML-header"
  >
    <div
      className="ML-title"
    >
      Group Members
    </div>
  </div>
  <hr
    className="ML-hr"
  />
</div>
`;
